### ISSUE TITLE
Environment Secrets have no value

### ISSUE THREAD

**Issue by pjarnfelt** [firebase/firebase-functions#1264]
### [REQUIRED] Version info

**node:** v16.17.1

**firebase-functions:** 4.0.0

**firebase-tools:** 11.4.4

**firebase-admin:** 11.1.0

### [REQUIRED] Test case
Firebase init
install latest version of admin/tools and functions.  (as my versions)
import defineSecret 
`import { defineSecret } from "firebase-functions/params";`
define functions to run with that secret
`runWith({secrets:[my_secret]})`
try to get the value inside the function

### [REQUIRED] Steps to reproduce

```js
import * as functions from "firebase-functions";
import { defineSecret } from "firebase-functions/params";

// Defined in GCloud or not, we just don't have access to its value!
const my_secret = defineSecret("my_secret");
// Remember to define runWith the secrets, (here only one in the array) 
// as it should only be pupulated at runtime from GCloud
export const helloWorld = functions.runWith({secrets:[my_secret]}).https.onRequest((request, response) => {
  my_secret.value(); // Property 'value' does not exist on type 'SecretParam'.ts(2339)
});
```
now
`my_secret.value();`
is undefined. SecretParam (the thing defineSecret returns) only has a .name parameter, no .value()

### [REQUIRED] Expected behavior

Expected to be able to reference .value() method as per the documentation
https://firebase.google.com/docs/functions/config-env

### [REQUIRED] Actual behavior
cannot access the value :(
**Property 'value' does not exist on type 'SecretParam'.ts(2339)*

### Were you able to successfully deploy your functions?

no

**Comment by jhuleatt** [comment]
Thanks for reporting this @pjarnfelt! I believe a temporary workaround could be to access the underlying environment variable directly:

```diff
import * as functions from "firebase-functions";
import { defineSecret } from "firebase-functions/params";

// Defined in GCloud or not, we just don't have access to its value!
const my_secret = defineSecret("my_secret");
// Remember to define runWith the secrets, (here only one in the array) 
// as it should only be pupulated at runtime from GCloud
export const helloWorld = functions.runWith({secrets:[my_secret]}).https.onRequest((request, response) => {
-  my_secret.value(); // Property 'value' does not exist on type 'SecretParam'.ts(2339)
+  const mySecretVal = process.env[my_secret.name]; 
});
```

@Berlioz, unlike the other param types (`StringParam`, `IntParam`, etc), `SecretParam` doesn't extend `Param`, so it doesn't pick up the `value` method from the `Expression` class. Therefore, at the moment it seems the only way to access the value of a secret param is to manually check the environment variable.

String: https://github.com/firebase/firebase-functions/blob/5c038e602b865265b903a89747972e4cae581960/src/params/types.ts#L364

Secret: https://github.com/firebase/firebase-functions/blob/5c038e602b865265b903a89747972e4cae581960/src/params/types.ts#L338

**Comment by jhuleatt** [comment]
The fix for this was released in [`4.0.2`](https://github.com/firebase/firebase-functions/releases/tag/v4.0.2)
